cassovary-core/src/main/scala/com/twitter/cassovary/graph/UndirectedGraph.scala

```diff
@@ -0,0 +1,29 @@
+package com.twitter.cassovary.graph
+
+import com.twitter.cassovary.graph.StoredGraphDir.StoredGraphDir
+
+/*
+ * A graph where the outbound neighbors of each node equal the inbound neighbors.
+ * Given a DirectedGraph which stores the neighbors of each node (as outbound edges),
+ * this class wraps that graph in an undirected view.
+ */
+
+class UndirectedGraph(outboundGraph: DirectedGraph) extends DirectedGraph {
+  override def getNodeById(id: Int): Option[Node] = outboundGraph.getNodeById(id) map { new UndirectedNode(_)}
+
+  override def iterator: Iterator[Node] = outboundGraph.iterator map { new UndirectedNode(_)}
+
+  override def edgeCount: Long = outboundGraph.edgeCount
+
+  override val storedGraphDir: StoredGraphDir = outboundGraph.storedGraphDir
+
+  override def nodeCount: Int = outboundGraph.nodeCount
+}
+
+class UndirectedNode(outboundNode: Node) extends Node {
```

Comment:
look at existing UnidirectionalNode.scala

UndirectedNode is a UnidirectionalNode with in and out bonded

so I'd put this inside UnidirectionalNode.scala

trait UndirectedNode extends UnidirectionalNode {
def inboundNodes() = outboundNodes()
}

and modify that scala file's objects' apply methods to return UndirectedNode when the direction is Mutual

while at it, please make UnidirectionalNode a trait instead of an abstract class.